File: packages/python-sdk/src/treedb_sdk/conformance/adapter.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Literal

from treedb_sdk.client import TreeDbClient


@dataclass(frozen=True)
class TreeDbConformanceScenario:
    id: str
    capability_id: str
    title: str
    required: bool
    endpoint_refs: list[str]
    steps: list[dict[str, str]]
    assertions: list[str]


@dataclass(frozen=True)
class TreeDbConformanceResult:
    scenario_id: str
    status: Literal["passed", "failed", "not_configured"]
    message: str | None = None


class TreeDbConformanceAdapter:
    def __init__(self, client: TreeDbClient, server_configured: bool = False) -> None:
        self.client = client
        self.server_configured = server_configured

    def run_scenario(self, scenario: TreeDbConformanceScenario) -> TreeDbConformanceResult:
        if not self.server_configured:
            return TreeDbConformanceResult(
                scenario_id=scenario.id,
                status="not_configured",
                message="TreeDB conformance server is not configured.",
            )
        return TreeDbConformanceResult(
            scenario_id=scenario.id,
            status="not_configured",
            message="Executable Python conformance dispatch is deferred to a later phase.",
        )

File: packages/elixir-sdk/lib/tree_db_sdk/conformance/adapter.ex
defmodule TreeDbSdk.Conformance.Adapter do
  @moduledoc false
  defstruct [:client, server_configured: false]

  def new(client, opts \\ []),
    do: %__MODULE__{
      client: client,
      server_configured: Keyword.get(opts, :server_configured, false)
    }

  def run_scenario(%__MODULE__{server_configured: configured}, scenario) do
    message =
      if configured,
        do: "executable scenario dispatch is deferred to a later phase",
        else: "TreeDB server is not configured"

    %{scenario_id: scenario["id"], status: :not_configured, message: message}
  end
end
